### PR TITLE
Added Categorical distribution notebook.

### PR DESCRIPTION
**Pull request by rohit-khoiwal**
### **Steps** :

- Added categorical_distribution.ipynb to notebooks/introduction.
- Added softmax in gradient descent for parameters to add up to ones.

### CODE REVIEW

File: notebooks/introduction/categorical_distribution.ipynb [hunk 1]
```diff
@@ -0,0 +1,656 @@
+{
```
Comment (patel-zeel) [inline notebooks/introduction/categorical_distribution.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #4.    import optax
</pre><p>We can adapt to same try: except: pattern as we use for pml books.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/code-first-ml/book1/pull/3/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='5'/>

Comment (patel-zeel) [inline notebooks/introduction/categorical_distribution.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #1.    theta = [1/6]*6
</pre><p>Would you like to mention just above this what is the interpretation of this theta? For example, Die is unbiased here because all probabilities are same.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/code-first-ml/book1/pull/3/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='6'/>

Comment (patel-zeel) [inline notebooks/introduction/categorical_distribution.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #5.    count_cat_samples
</pre><p>A histogram would be better.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/code-first-ml/book1/pull/3/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='8'/>

Comment (patel-zeel) [inline notebooks/introduction/categorical_distribution.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #1.    import matplotlib.pyplot as plt
</pre><p>We can keep all imports in first cell to avoid distraction in between.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/code-first-ml/book1/pull/3/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='11'/>

Comment (patel-zeel) [inline notebooks/introduction/categorical_distribution.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #2.        d = 1 - jnp.sum(jnp.array([a,b,c]))
</pre><p>more meaningful variable names might be used, if not, you can explain what is happening in the markdown cell just above this.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/code-first-ml/book1/pull/3/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='12'/>

Comment (patel-zeel) [inline notebooks/introduction/categorical_distribution.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #11.            for i in range(len(self.thetas)):
</pre><p>You can update self.thetas directly without using a for loop or .at method here. I think you are not using this function anywhere so removing it would be good.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/code-first-ml/book1/pull/3/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='20'/>

Comment (patel-zeel) [inline notebooks/introduction/categorical_distribution.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #15.    print("Theta's values by MLE\t:",mle_thetas)
</pre><p>It might be useful to show upto 2 or 3 decimal places to show small differences between true values and MLE. You can use f-string formatting instead of round method here.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/code-first-ml/book1/pull/3/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='28'/>

Comment (rohit-khoiwal) [reply notebooks/introduction/categorical_distribution.ipynb:1]: It's because we are taking 10000 samples.
